client/v2/autocli/flag/map_test.go

```diff
@@ -0,0 +1,142 @@
+package flag
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+type mockValue struct {
+	val string
+	err error
+}
+
+func (m *mockValue) Set(s string) error {
+	if m.err != nil {
+		return m.err
+	}
+	m.val = s
+	return nil
+}
+
+func (m *mockValue) String() string {
+	return m.val
+}
+
+func (m *mockValue) Type() string {
+	return "mock"
+}
+
+func (m *mockValue) Get(_ protoreflect.Value) (protoreflect.Value, error) {
+	if m.err != nil {
+		return protoreflect.Value{}, m.err
+	}
+	return protoreflect.ValueOfString(m.val), nil
+}
+
+type mockType struct {
+	err error
+}
+
+func (m *mockType) NewValue(ctx *context.Context, opts *Builder) Value {
+	return &mockValue{err: m.err}
+}
+
+func (m *mockType) DefaultValue() string {
+	return ""
+}
+
+func TestCompositeMapValue_Set(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		resolver   keyValueResolver[int]
+		valueType  Type
+		expectErr  bool
+		expectVals map[int]string
+	}{
+		{
+			name:  "valid input",
+			input: "1=foo,2=bar",
+			resolver: func(s string) (int, error) {
+				switch s {
+				case "1":
+					return 1, nil
+				case "2":
+					return 2, nil
+				default:
+					return 0, errors.New("invalid key")
+				}
+			},
```

Comment:
```suggestion
			resolver: strconv.Atoi,
```

Reply:
this works. i think we should either define all resolvers as strconv.Atoi, or remove the field from the test case and force it to be strconv.Atoi.